src/game/Entities/Unit.cpp

```diff
@@ -5856,36 +5856,36 @@ void Unit::CasterHitTargetWithSpell(Unit* realCaster, Unit* target, SpellEntry c
         if (spellInfo->HasAttribute(SPELL_ATTR_EX3_NO_INITIAL_AGGRO) && !spellInfo->HasAttribute(SPELL_ATTR_EX3_OUT_OF_COMBAT_ATTACK))
             return;
 
+        // we want to change the stand state of each character if possible/required
+         // Since patch 1.5.0 sitting creature always stand up on attack (even if stunned)
+        if (success)
+        {
+            if (!spellInfo->HasAttribute(SPELL_ATTR_CASTABLE_WHILE_SITTING) && !realCaster->IsStandState())
+                realCaster->SetStandState(UNIT_STAND_STATE_STAND);
```

Comment:
I want to point out that in the old code we did not call this anywhere.

Reply:
yeah I know, I included it because the old code seemed like it had messed up thee logic. The SPELL_ATTR_CASTABLE_WHILE_SITTING was checked to see if it could make the target stand. Figured it was mean't for the caster given its not used anywhere else in the codebase.

Reply:
Actually, I have different evidence. In the classic beta, the spells without the attribute are not casted. You get a message "you need to be standing". Which leads me to believe this code is not intended.

Reply:
So rather than force standing, we want to perform a spell check for our standing state and reject it if this attribute doesn't exist?

Reply:
For caster, yes. For target keep the reworked code.